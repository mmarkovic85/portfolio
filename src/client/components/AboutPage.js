import React from "react";

const AboutPage = () => (
  <h1>About page placeholder</h1>
);

export {
  AboutPage as default
};
