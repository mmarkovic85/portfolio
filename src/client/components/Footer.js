import React from "react";

const Footer = () => (
  <footer>
    <h6>Footer placeholder</h6>
  </footer>
);

export {
  Footer as default
};
