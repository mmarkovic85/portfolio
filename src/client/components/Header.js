import React from "react";

const Header = () => (
  <header>
    <h6>Header placeholder</h6>
  </header>
);

export {
  Header as default
};
