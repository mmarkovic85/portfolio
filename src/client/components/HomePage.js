/*
 * HomePage Component
 *
 * For testing purposes divided in two sub-components:
 *  ContextConnected - connect HomePage to application context
 *  HomePage - rendering Component
 */
import React, { useContext } from "react";
import { AppContext } from "../context/AppContext";

// Render HomePage Component
const HomePage = ({ title }) => (
  <div>
    <h1 className="homepage__title">Welcome to {title}.</h1>
  </div>
);

// Connect HomePage to application context
const ContextConnected = () => {
  const { appTitle } = useContext(AppContext);

  return (<HomePage title={appTitle} />);
}

export {
  HomePage,
  ContextConnected as default
};
