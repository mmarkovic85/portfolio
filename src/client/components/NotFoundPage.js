/*
 * NotFoundPage Component
 *
 * For testing purposes divided in two sub-components:
 *  ContextConnected - connect NotFoundPage to application context
 *  NotFoundPage - rendering Component
 */
import React, { useContext } from "react";
import { AppContext } from "../context/AppContext";

// Render NotFoundPage Component
const NotFoundPage = ({ url }) => (
  <div>
    <h1 className="not-found-page__title">Error 404</h1>
    <p>The requested URL "{url}" was not found!</p>
  </div>
);

// Connect NotFoundPage to application context
const ContextConnected = ({ location: { pathname } }) => {
  const { appDomainAddress } = useContext(AppContext);

  return (
    <NotFoundPage
      url={appDomainAddress + pathname}
    />
  );
}

export {
  NotFoundPage,
  ContextConnected as default
};
