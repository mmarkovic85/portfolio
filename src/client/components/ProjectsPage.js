import React from "react";

const ProjectsPage = () => (
  <h1>Projects page placeholder</h1>
);

export {
  ProjectsPage as default
};
