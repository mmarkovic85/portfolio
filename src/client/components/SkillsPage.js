import React from "react";

const SkillsPage = () => (
  <h1>Skills page placeholder</h1>
);

export {
  SkillsPage as default
};
