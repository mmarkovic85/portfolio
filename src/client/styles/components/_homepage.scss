/*
 * HomePage Component stylesheet
 */

.homepage__title {
  color: $red;
}
