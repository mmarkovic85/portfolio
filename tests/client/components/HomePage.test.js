/*
 * HomePage Component test file
 */
import React from "react";
import renderer from "react-test-renderer";
import { HomePage } from "../../../src/client/components/HomePage";

test("Should render HomePage component correctly", () => {
  // render HomePage Component
  const rendered = renderer.create(
    <HomePage title={"My App"} />
  );

  // Save/match snapshot of/to Component
  expect(rendered.toJSON()).toMatchSnapshot();
});
