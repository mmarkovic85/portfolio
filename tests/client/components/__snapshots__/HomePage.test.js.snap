// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render HomePage component correctly 1`] = `
<div>
  <h1
    className="homepage__title"
  >
    Welcome to 
    My App
    .
  </h1>
</div>
`;
